drivers/flash/flash_shell.c

```diff
@@ -354,6 +354,30 @@ static int read_write_erase_validate(const struct shell *sh, size_t argc, char *
 	return 0;
 }
 
+static int erase_validate(const struct shell *sh, size_t argc, char *argv[],
```

Comment:
Wouldn't this be better to set this as "common validation" and do the extra
```
*size > CONFIG_FLASH_SHELL_BUFFER_SIZE
```
check for read/write.
Current solution duplicates code.